src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityFluidHatch.java

```diff
@@ -70,8 +83,33 @@ public class MetaTileEntityFluidHatch extends MetaTileEntityMultiblockNotifiable
     public MetaTileEntityFluidHatch(ResourceLocation metaTileEntityId, int tier, boolean isExportHatch) {
         super(metaTileEntityId, tier, isExportHatch);
         this.fluidTank = new HatchFluidTank(getInventorySize(), this, isExportHatch);
+        initializeInventory(); // the fact that this has to be called three times is so dumb
         this.workingEnabled = true;
-        initializeInventory();
+    }
+
+    @Override
+    protected void initializeInventory() {
+        super.initializeInventory();
+        if (this.hasGhostCircuitInventory()) {
+            this.circuitInventory = new GhostCircuitItemStackHandler(this);
+            this.circuitInventory.addNotifiableMetaTileEntity(this);
```

Comment:
Why do we need to add the fluid hatch to the notified list? Since the hatch can never perform any recipes. We get the multiblock in the add to structure override, but I don't think there is any need for the hatch itself to be notified, is there?

Reply:
idk, every other usage of `GhostCircuitItemStackHandler` does this as well, but it doesn't seem to be needed?